vissl/engines/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,545 @@
+        while True:
+            if time.time() - start_time > _TRAINING_CHECKPOINT_FILE_WAIT_SECONDS:
+                raise RuntimeError(
+                    f"Training checkpoint folder still doesn't exist after:"
+                    f"{_TRAINING_CHECKPOINT_FILE_WAIT_SECONDS / 60} minutes"
+                )
+
+            if PathManager.exists(self.training_checkpoint_file):
+                # Load training yaml config.
+                self._load_training_config()
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+        self.save_evaluation_results()
+
+        # Checkpoint folder is now available. Continuously monitor the training checkpoints,
+        # launch evaluation jobs as needed, monitor their progress, and record their results.
+        while True:
+            self._evaluate_checkpoints()
+
+            self._check_evaluation_jobs()
+
+            # Break if no more checkpoints to evaluate
+            if self._finished():
+                print("Evaluations are finished")
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+    def save_evaluation_results(self):
+        """
+        Upload json file to the evaluation directory.
+        """
+        # Upload all checkpoints evaluations to parent checkpoint directory.
+        evaluation_dir = self.evaluation_dir()
+        parent_metrics_file = os.path.join(evaluation_dir, "evaluation_metrics.json")
+        Path(evaluation_dir).mkdir(parents=True, exist_ok=True)
+        self._write_json_file(self.evaluation_results, parent_metrics_file)
+
+        # Upload each checkpoint's evaluations to child directories.
+        for checkpoint_str, benchmarks in self.evaluation_results.items():
+            child_metrics_dir = os.path.join(evaluation_dir, checkpoint_str)
+            child_metrics_file = os.path.join(
+                child_metrics_dir, "evaluation_metrics.json"
+            )
+            Path(child_metrics_dir).mkdir(parents=True, exist_ok=True)
+            self._write_json_file(benchmarks, child_metrics_file)
+
+        print("Saved evaluation Results")
+
+    def evaluation_dir(self):
+        return os.path.join(self.training_checkpoint_dir, "evaluations")
+
+    def _load_training_config(self):
+        # Load training yaml config.
+        self.training_config = load_file(self.training_checkpoint_file)
+        self.training_config = AttrDict(self.training_config)
+        print(
+            f"Loaded training checkpoint config from: { self.training_checkpoint_file }"
+        )
+        # Build main training task in order to extract iteration info.
+        self.training_task = build_task(self.training_config)
+        self.training_task.dataloaders = self.training_task.build_dataloaders(False)
+        self.evaluation_results = self._generate_initial_evaluation_results()
+        self._validate_training_cfg()
+
+    def _validate_class_instance(self):
```

Comment:
question: it sounds like we are simply checking availability of libraries like hydra, submitit and nothing to do with the class itself. Could we find a better place for these functions and possible just call them directly instead of putting them under the `_validate_class_instance` ? :)